### PR TITLE
Update std library section

### PR DESCRIPTION
**Pull request by bravit**
- restructure lessons
- smaller steps
- a couple of new tasks (Filter a Sequence, Switch the Lamp)
- prefixing checkable tasks with "Task:"

### CODE REVIEW

**Review by sofiiako — APPROVED** [review]

File: Standard Library Types/Closures/Filter a sequence with a closure/task.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+## Task: Filter a Sequence with a Closure
+
+You have an array with `i32` elements, a `cap`, and a counter `left_behind`. Your task
+is to complete the closure given as a parameter to the `filter` method. The closure
+should keep only elements, strictly smaller than `cap`. It should also count all the elements
```
Comment (stephen-hero) [inline Standard Library Types/Closures/Filter a sequence with a closure/task.md:5]: should keep only the elements strictly smaller than

File: Standard Library Types/Closures/Filter a sequence with a closure/task.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+left behind.
+
+<div class="hint">
+Read the docs about the <code>filter</code> method to understand what are
```
Comment (stephen-hero) [inline Standard Library Types/Closures/Filter a sequence with a closure/task.md:9]: Read the docs about the <code>filter</code> method to understand what

File: Standard Library Types/Closures/Filter a sequence with a closure/task.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+## Task: Filter a Sequence with a Closure
+
+You have an array with `i32` elements, a `cap`, and a counter `left_behind`. Your task
+is to complete the closure given as a parameter to the `filter` method. The closure
+should keep only elements, strictly smaller than `cap`. It should also count all the elements
+left behind.
+
+<div class="hint">
+Read the docs about the <code>filter</code> method to understand what are 
+those <code>&&</code> in the closure argument about.
```
Comment (stephen-hero) [inline Standard Library Types/Closures/Filter a sequence with a closure/task.md:10]: those <code>&&</code> in the closure argument are about.

File: Standard Library Types/Closures/Filter a sequence with a closure/task.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+</div>
+
+<div class="hint">
+You may want to use the fact that the right hand side of <code> condition || condition </code> is
```
Comment (stephen-hero) [inline Standard Library Types/Closures/Filter a sequence with a closure/task.md:14]: right-hand side

File: Standard Library Types/Smart Pointers/Use Rc and RefCell/task.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+## Switch the Lamp
+
+In this task, you have a single lamp, and several external switchers which are able to operate on the lamp. To make this happen we need both `Rc` and `RefCell` smart pointers. `Rc` allows us to have multiple owners, and `RefCell` provides mutability with `borrow_mut`. 
+
+We'd like to implement the following scenario:
+
+* create a lamp
+* create a switcher for a lamp
```
Comment (stephen-hero) [inline Standard Library Types/Smart Pointers/Use Rc and RefCell/task.md:8]: create a switcher for the lamp

File: Standard Library Types/Smart Pointers/Use Rc and RefCell/task.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+## Switch the Lamp
+
+In this task, you have a single lamp, and several external switchers which are able to operate on the lamp. To make this happen we need both `Rc` and `RefCell` smart pointers. `Rc` allows us to have multiple owners, and `RefCell` provides mutability with `borrow_mut`. 
+
+We'd like to implement the following scenario:
+
+* create a lamp
+* create a switcher for a lamp 
+* create another switcher for the same lamp 
+* operate on different switchers and make sure that the lamp is on, and then off again
```
Comment (stephen-hero) [inline Standard Library Types/Smart Pointers/Use Rc and RefCell/task.md:10]: lamp is on and then off

File: Standard Library Types/Smart Pointers/Use Rc and RefCell/task.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+In fact, this scenario is already implemented in `main.rs`. Edit the `switchers` module by following the hints there to make the code compile.
+
+<div class="hint">Read the main function to understand which type should we use for <code>lamp</code> in <code>Switcher</code>.</div>
```
Comment (stephen-hero) [inline Standard Library Types/Smart Pointers/Use Rc and RefCell/task.md:14]: to understand which type we should use

File: Standard Library Types/Smart Pointers/Use Rc and RefCell/task.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+<div class="hint">Note the <code>&</code> in the call to <code>Switcher::new</code>.</div>
+
+<div class="hint">To clone the reference, you may need <code>Rc::clone</code>.</div>
+
+<div class="hint"><code>RefCell</code> gives you <code>borrow</code> and <code>borrow_mut</code> methods. When should you use them?</div>
```
Comment (stephen-hero) [inline Standard Library Types/Smart Pointers/Use Rc and RefCell/task.md:20]: gives you the

**Pull request merged**